oak_channel/SPEC.md

```diff
@@ -138,149 +93,80 @@ A byte encoded frame MUST consist of the following fields:
  0                   1                   2                   3
  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                            padding                            |
+|                         invocation_id                         |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                         message_length                        |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|             length            |             flags             |
+|                          frame_number                         |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                          body_length                          |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 |                                                               |
 +                         body bytes...                         +
 |                                                               |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ ...
 ```
 
-<!-- Diagram generated with https://www.luismg.com/protocol/, using the schema
-"padding:32,length:16,flags:16,body bytes...:64" -->
+<!-- Diagram generated with https://www.luismg.com/protocol/, using the spec
+"invocation_id:32,message_length:32,frame_number:32,body_length:32,body bytes...:64" -->
 
-- `padding`, 4 bytes
+- `invocation_id`, u32, little endian
```

Comment:
Instead of specifying that all these fields are "little endian", I'd propose "machine native endianness" -- these messages will be exchanged between processes on the same machine, so if we ever move to a BE machine having to do two swaps seems a bit silly.

Reply:
Does the restricted kernel's endiannes also depend on the architecture?

Reply:
Yes, on a big-endian CPU we'd run the kernel in big-endian mode as well.

Reply:
I disagree. Communication protocol specifications are usually explicit about endianness to avoid doubt. Also, at the moment we are running the untrusted launcher and the host on the same machine, but that might not always be the case so we cannot assume that we will always have the same endianness on both sides of the channel (even though it seems extremely unlikely that we will ever run either part on a BE machine).